DMCompiler/DM/DMExpression.cs

```diff
@@ -138,10 +138,10 @@ private void VerifyArgType(DMCompiler compiler, DMProc targetProc, int index, st
     }
 }
 
-internal readonly struct ExpressionContext(DMCompiler compiler, DMObject type, DMProc proc) {
+internal readonly struct ExpressionContext(DMCompiler compiler, DMObject type, DMProc? proc) {
     public readonly DMCompiler Compiler = compiler;
     public readonly DMObject Type = type;
-    public readonly DMProc Proc = proc;
+    public readonly DMProc? Proc = proc;
```

Comment:
This introduces a million nullability warnings elsewhere. There are only 3 cases where an ExpressionContext is created with a null `proc` argument. I think it would be better to instead provide the type's `<init>` proc for those.

I'd prefer that to be done in another PR though.